### PR TITLE
Add CSV and JSON export utilities for PSLab sensor data

### PR DESCRIPTION
**Pull request by saadiqbal09**
This PR adds utility functions to export PSLab sensor data
to CSV and JSON formats.

- Useful for IoT data logging and analysis
- Pure Python implementation
- Includes basic pytest-based tests
- No hardware required

This improves data usability for PSLab users.

## Summary by Sourcery

Add utilities for exporting PSLab sensor data to common file formats and ensure they are covered by basic tests.

New Features:
- Introduce helpers to export PSLab sensor data to CSV and JSON files.

Tests:
- Add pytest-based tests verifying CSV and JSON export create output files and that exporting empty data raises an error.

### CODE REVIEW

**Comment by saadiqbal09** [comment]
This is my first contribution to PSLab.
Happy to update the implementation based on feedback.

File: tests/test_data_export.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import pytest
+from pslab.utils.data_export import export_to_csv, export_to_json
+
+def test_export_csv_creates_file(tmp_path):
+    data = [{"a": 1, "b": 2}]
+    file_path = tmp_path / "data.csv"
+    export_to_csv(data, str(file_path))
+    assert file_path.exists()
```
[the remark this answers is not in the document]
Comment (saadiqbal09) [reply tests/test_data_export.py]: Thanks for the detailed suggestions!

I’ve updated the tests to validate the actual CSV and JSON contents
(including headers, row counts, and parsed values) as recommended.
Please let me know if anything else should be improved.

**Comment by mariobehling** [comment]
Small process note.

We have automatic Copilot PR reviews enabled on this repository. These reviews are only triggered if the contributor has GitHub Copilot enabled and an active license on their own account.

Please enable Copilot in your GitHub settings if you have access. In many regions, free licenses are available through educational institutions or developer programs. Enabling Copilot helps us speed up the auto review process and reduces manual review overhead for the core team.

The team will review your PR. Thank you for your contribution and cooperation.

**Review by Saksham-Sirohi — COMMENTED** [review]
Copilot flagged the following:

- Tests-only companion to #260 — CI fails if merged alone.
